### PR TITLE
Handling api key

### PR DESCRIPTION
**Pull request by kosodrzewina**
Fixed problems related to an invalid API key (or lack of one):
- if there is no API key search field, channel ID button and remove API key button are disabled
- if user searches with an invalid API key, infinite loading bug does not longer occur

https://user-images.githubusercontent.com/34711852/131897629-c5324444-2436-4684-85a6-b5adff8487fa.mp4

### CODE REVIEW

File: lib/pages/api_key_dialog.dart [hunk 1]
```diff
@@ -93,24 +94,45 @@ class _ApiKeyDialogState extends State<ApiKeyDialog> {
...
-
-              widget.profileScaffoldMessengerKey.currentState!
-                ..hideCurrentSnackBar()
-                ..showSnackBar(
-                  AppSnackBar.success(
-                    content: Text(
-                      L10n.of(context)!.apiKeyDialog_apiKeyRemoved,
+          Observer(
+            builder: (_) => IconTextButton(
+              icon: context.read<AuthStore>().apiKey != null
```
Comment (shilangyu) [inline lib/pages/api_key_dialog.dart]: Maybe calculate hasApiKey once?

File: lib/widgets/icon_text_button.dart [hunk 2]
```diff
@@ -3,8 +3,8 @@ import 'package:flutter/material.dart';
 
 class IconTextButton extends StatefulWidget {
   final Icon icon;
-  final String text;
-  final Function() onPressed;
+  final Text text;
+  final Function()? onPressed;
```
Comment (shilangyu) [inline lib/widgets/icon_text_button.dart]: `VoidCallback?` :>

File: lib/pages/search_page.dart [hunk 3]
```diff
@@ -15,28 +19,86 @@ class SearchPage extends StatefulWidget {
...
+                    style: TextStyle(
+                      color: Theme.of(context)
+                          .textTheme
+                          .bodyText1!
+                          .color!
+                          .withOpacity(0.3),
+                    ),
+                  ),
+            onPressed: context.read<AuthStore>().apiKey != null
+                ? () async => await showChannelIdDialog(context)
```
Comment (shilangyu) [inline lib/pages/search_page.dart]: async/await not needed

File: lib/pages/search_page.dart [hunk 4]
```diff
@@ -47,4 +109,10 @@ class _SearchPageState extends State<SearchPage> {
       ],
     );
   }
+
+  @override
+  void dispose() {
```
Comment (shilangyu) [inline lib/pages/search_page.dart:116]: controller also has to be disposed

File: lib/widgets/icon_text_button.dart [hunk 5]
```diff
@@ -3,8 +3,8 @@ import 'package:flutter/material.dart';
 
 class IconTextButton extends StatefulWidget {
   final Icon icon;
-  final String text;
-  final Function() onPressed;
+  final Text text;
```
Comment (shilangyu) [inline lib/widgets/icon_text_button.dart]: I would just make it a `Widget`

**Review by shilangyu — CHANGES_REQUESTED** [review]

File: lib/widgets/icon_text_button.dart [hunk 6]
```diff
@@ -3,13 +3,13 @@ import 'package:flutter/material.dart';
 
 class IconTextButton extends StatefulWidget {
   final Icon icon;
-  final String text;
-  final Function() onPressed;
+  final Widget widget;
```
Comment (shilangyu) [inline lib/widgets/icon_text_button.dart]: `text` was ok, but `child` is also ok, `widget` is not ok

File: lib/pages/api_key_dialog.dart [hunk 7]
```diff
@@ -93,25 +94,48 @@ class _ApiKeyDialogState extends State<ApiKeyDialog> {
             ],
           ),
           const SizedBox(height: 10),
-          IconTextButton(
-            icon: const Icon(Icons.delete),
-            text: L10n.of(context)!.apiKeyDialog_removeApiKey,
-            onPressed: () {
-              context.read<AuthStore>().setApiKey(null);
+          Observer(builder: (_) {
+            final isApiKey = context.read<AuthStore>().apiKey != null;
```
Comment (shilangyu) [inline lib/pages/api_key_dialog.dart:98]: `isApiKey` or `hasApiKey`?

**Review by shilangyu — APPROVED** [review]

File: lib/pages/api_key_dialog.dart (hunk 7 above)
Comment (kosodrzewina) [reply lib/pages/api_key_dialog.dart:98]: Why should it be hasApiKey?

**Pull request merged**